community/boilerplates/jwt-authentication-server/controllers/user.js

```diff
@@ -0,0 +1,81 @@
+const passport = require('../config/passport');
+const { User } = require('../db/schema');
+const { errorHandler } = require('../db/errors');
+const jwt = require('jsonwebtoken');
+
+/**
+ * POST /login
+ * Sign in using username and password.
+ */
+exports.postLogin = async (req, res, next) => {
+  req.assert('username', 'Username is not valid').notEmpty();
+  req.assert('password', 'Password cannot be blank').notEmpty();
+
+  const errors = req.validationErrors();
+
+  if (errors) {
+    return res.status(400).json({'errors': errors});
+  }
+
+  passport.authenticate('local', (err, user) => {
+    if (err) { return handleResponse(res, 400, {'error': err})}
+    if (user) {
+
+      const tokenContents = {
+        sub: '' + user.id,
+        name: user.username,
+        iat: Date.now() / 1000,
+        "https://hasura.io/jwt/claims": {
+          "x-hasura-allowed-roles": ["editor","user", "mod"],
+          "x-hasura-user-id": '' + user.id,
+          "x-hasura-default-role": "user",
+          "x-hasura-role": "user"
+        }
+      }
+
+      handleResponse(res, 200, {
+        token: jwt.sign(tokenContents, process.env.ENCRYPTION_KEY)
+      });
+    }
+  })(req, res, next);
+};
+
+
+/**
+ * POST /signup
+ * Create a new local account.
+ */
+exports.postSignup = async (req, res, next) => {
+  req.assert('username', 'Username is not valid').notEmpty();
+  req.assert('password', 'Password must be at least 4 characters long').len(4);
+  req.assert('confirmPassword', 'Passwords do not match').equals(req.body.password);
+
+  const errors = req.validationErrors();
+
+  if (errors) {
+    return res.status(400).json({'errors': errors});
+  }
+
+  try {
+    await User.query()
+                           .allowInsert('[username, password]')
+                           .insert({
+                             username: req.body.username,
+                             password: req.body.password
```

Comment:
Shouldn't this password be salted/hashed and saved. Maybe use a standard password creating library? This library will be used during signup for inserting, and comparing passwords when logging in?